kong/runloop/handler.lua

```diff
@@ -1389,9 +1389,7 @@ return {
       -- We overcome this behavior with our own logic, to preserve user
       -- desired semantics.
       -- perf: branch usually not taken, don't cache var outside
-      if byte(ctx.request_uri or var.request_uri, -1) == QUESTION_MARK then
-        var.upstream_uri = var.upstream_uri .. "?"
-      elseif var.is_args == "?" then
+      if var.is_args == "?" or byte(ctx.request_uri, -1) == QUESTION_MARK then
```

Comment:
```suggestion
      if byte(ctx.request_uri or var.request_uri, -1) == QUESTION_MARK or var.is_args == "?" then
```
as mentioned by @oowl it should probably still include the fallback to `var.request_uri` as a source of truth, just in case the ctx variable was unset for some reason

Reply:
Works. 

I have also made the changes suggested here. Added var.request_uri as a fallback, and revered the order of var.is_args in the boolean expression.

Reply:
> Could you highlight any case where ctx.request_uri may be unset for any reason

This may not be possible right now, it is just a safety measure